[DOC] Document role restriction for API keys (#97027)
elastic/elasticsearch: x-pack/docs/en/security/authorization/role-restriction.asciidoc

Author: Thanks for the suggestion and offer. I've [tagged the example](https://github.com/elastic/elasticsearch/pull/97027/files#diff-8930c0cbd8a015aeb5dfcfcba987a0022547bad58969485af96c929f5b0ca16dR215-R239) and [included at the end](https://github.com/elastic/elasticsearch/pull/97027/files#diff-8719c5d0669c504228def012e0ccce513904789289f59ae187f7231062fd9d36R31) of this file. It was straightforward.

[role="xpack"]
[[role-restriction]]
=== Role restriction

Role restriction can be used to specify conditions under which a role should be effective.
Not specifying restriction means a role is not restricted and thus always effective.

> This is the default behaviour.

Reviewer: We should also describe here the behaviour when restriction is specified but the condition is not met, i.e. when the role is effectively disabled. I am aware this is described in the following "Workflows" section. But I think it makes sense to have a overall statement here at the Restriction level. At workflows level, we can be more specific on what it means by not matching the condition, i.e. not calling the allowed REST APIs. I hope this makes sense.

Author: Makes sense. I've tried to reformulate it, but also not repeat myself. Let me know if it reads better now.

--
NOTE: Currently, the role restriction is only supported for <<security-api-create-api-key, API keys>>,

> with limitation that it's allowed to specify only a single role descriptor with restriction.

Reviewer: ```suggestion
with limitation that the API key can only have a single role descriptor.
```

--

[[workflows-restriction]]
==== Workflows

Workflows allow to restrict the role to be effective exclusively when calling certain REST APIs.
Calling a REST API that is not allowed by a workflow, will result in the role being disabled (i.e. _empty_) and thus access being denied.
The below section lists workflows that you can restrict the role to:

`search_application_query`::: This workflow restricts the role to the <<search-application-search, Search Application Search API>> only.

--

> NOTE: Workflows are case-sensitive.

Reviewer: Nit: 
```suggestion
NOTE: Workflow names are case-sensitive.
```
